logstash-core/lib/logstash/environment.rb

```diff
@@ -2,6 +2,7 @@
 require "logstash/errors"
 require "logstash/config/cpu_core_strategy"
 require "logstash/settings"
+require "socket"
```

Comment:
OK got it why we need it. make sense!

https://github.com/elastic/logstash/pull/5511/files#diff-b803768250328614e95dee4f5b27af18R10